mongo/datadog_checks/mongo/mongo.py

```diff
@@ -216,10 +216,26 @@ def _check(self):
             tags.append('sharding_cluster_role:mongos')
 
         if isinstance(deployment, ReplicaSetDeployment) and deployment.is_arbiter:
+            self.log.debug("Replicaset and arbiter deployment, no databases will be checked")
             dbnames = []
         else:
-            dbnames = api.list_database_names()
-            self.gauge('mongodb.dbs', len(dbnames), tags=tags)
+            if self._config.db_names is None:
+                self.log.debug("Not configured databases. Retrieving list of databases on the mongo server")
+                dbnames = api.list_database_names()
+                self.gauge('mongodb.dbs', len(dbnames), tags=tags)
+            else:
+                self.log.trace("Collecting only from the configured databases: %s", self._config.db_names)
+                dbnames = []
+                server_databases = api.list_database_names()
+                self.gauge('mongodb.dbs', len(server_databases), tags=tags)
+                self.log.debug("Checking the configured databases that exist on the mongo server")
+                for config_dbname in self._config.db_names:
+                    if config_dbname in server_databases:
+                        self.log.trace("'%s' database found on the mongo server", config_dbname)
+                        dbnames.append(config_dbname)
+                    else:
+                        self.log.warning("'%s' database not found on the mongo server", config_dbname)
```

Comment:
Would it be worth adding a unit test for this, extracting this in a method and test only this part (and not the whole check)?